tests/mocks.py

```diff
@@ -131,10 +168,69 @@
     }
 }
 
+# One version of products named 'cos', 'sat' and 'cpe' that has valid YAML but not matching schema
+# - 'cos' product with invalid component manifests
+# - 'sat' product with invalid component docker
+# - 'cpe' product with invalid component s3
+MOCK_INVALID_PRODUCT_DATA = {
+    'cos': {
+        '2.1': {
+            'component_versions': {
+                'manifests': 'should be an array'
+            }
+        }
+    },
+    'sat': {
+        '2.1': {
+            'component_versions': {
+                'docker': [
+                    {'name': 'cray-product-catalog-update'}
+                ]
+            }
+        }
+    },
+    'cpe': {
+        '2.1': {
+            'component_versions': {
+                's3': [
+                    {'bucket': 'boot-images'}
```

Comment:
Maybe also add a check like this where `key` is missing instead of `bucket`.
And one where a non-string `key` or `bucket` is used, since both are required to be strings per the schema.